### PR TITLE
Update object states in hitl to use CollaborationSim

### PR DESCRIPTION
**Pull request by MatthewChang**
## Motivation and Context

Updates the HITL tool to use instances of CollaborationSim so that the latest object state logic is available in HITL. Depends on (https://github.com/facebookresearch/habitat-llm/pull/279)

## How Has This Been Tested

Manual local testing

## Types of changes
Development

## Checklist

- [x ] My code follows the code style of this project.
- [ ] I have updated the documentation if required.
- [ ] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes if required.

### CODE REVIEW

File: habitat-hitl/habitat_hitl/_internal/config_helper.py [hunk 1]
```diff
@@ -19,6 +21,14 @@ def update_config(
     debug_third_person_height=None,
 ):
     with habitat.config.read_write(config):  # type: ignore
+        # Move the metadata info to the habitat simulator config
+        if "metadata" in config:
```
Comment (MatthewChang) [inline habitat-hitl/habitat_hitl/_internal/config_helper.py:25]: Not sure if this repo has an existing solution for metadata.

File: habitat-hitl/habitat_hitl/environment/controllers/llm_controller.py [hunk 2]
```diff
@@ -208,12 +208,20 @@ def push_user_actions_to_llm(self):
             elif action["action"] == "PLACE":
                 furniture_name = "unknown furniture"
                 if action["receptacle_id"] is not None:
-                    receptacle_object = get_obj_from_id(self.environment_interface.sim, action["receptacle_id"])
-                    receptacle_handle = receptacle_object.handle if hasattr(receptacle_object, "handle") else None
+                    receptacle_object = get_obj_from_id(
```
Comment (MatthewChang) [inline habitat-hitl/habitat_hitl/environment/controllers/llm_controller.py:211]: linting

File: habitat-hitl/habitat_hitl/_internal/config_helper.py (hunk 1 above)
Comment (zephirefaith) [reply habitat-hitl/habitat_hitl/_internal/config_helper.py:25]: Can you check how are the affordances being read in the base code for this change?

**Review by zephirefaith — COMMENTED** [review]
Not sure if these changes are strictly required. Could you take a look at my questions? Thanks!

File: habitat-hitl/habitat_hitl/_internal/config_helper.py (hunk 1 above)
Comment (MatthewChang) [reply habitat-hitl/habitat_hitl/_internal/config_helper.py:25]: The only usages I see of metadata in phase3 are things like 

habitat-lab/habitat/datasets/rearrange/rearrange_generator.py
147:                self.sim.metadata_mediator.get_scene_handles(),
506:            mm=self.sim.metadata_mediator, scene_handle=ep_scene_handle

which is using the habitat_sim metadata, which is not what we need for the object states.

and 
habitat-hitl/habitat_hitl/_internal/networking/keyframe_utils.py
53:    # add or update instance metadata
54:    if "metadata" in inc_keyframe:
55:        ensure_list(consolidated_keyframe, "metadata")
56:        for metadata_container in inc_keyframe["metadata"]:
57:            key = metadata_container["instanceKey"]
58:            metadata = metadata_container["metadata"]
59:            if "metadata" in consolidated_keyframe:

Which appears to be a part of the networking.

I see no usages of "affordances", "affordance", or "afford" anywhere in phase3 in this repo other than usages in the object state machine code itself. Would you mind pointing me to the code you're referring to if I've somehow missed it?

**Comment by aclegg3** [comment]
I'm seeing the CI issue with habitat-llm being unavailable for import. We should probably instantiate that test on habitat-llm instead and mark it for skip here.

File: habitat-hitl/habitat_hitl/_internal/config_helper.py (hunk 1 above)
Comment (0mdc) [reply habitat-hitl/habitat_hitl/_internal/config_helper.py:25]: Added here: https://github.com/facebookresearch/habitat-lab/pull/2022

**Comment by 0mdc** [comment]
Closing as obsolete.

The following PRs implement this functionality:
* https://github.com/facebookresearch/habitat-lab/pull/2021
* https://github.com/facebookresearch/habitat-lab/pull/2022
* https://github.com/facebookresearch/habitat-lab/pull/2047

**Pull request closed without merging**